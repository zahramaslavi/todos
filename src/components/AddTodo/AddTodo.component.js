import React, { Component } from 'react';

import './AddTodo.component.scss';

class AddTodo extends Component {

    constructor(props) {
        super(props);

        this.state = {
            description: '',
            completed: false,
        };
    }

    handleSubmit(e) {
        this.props.addTodo(this.state);
        this.clearState();
    }

    clearState() {
        this.setState({
            description: '',
            completed: false
        })
    }

    render() { 
        return (
            <div className="addTodo">
                <form onSubmit={e => this.handleSubmit(e)}>
                    <input 
                        type="text" 
                        placeholder="Type to add new tasks"
                        value={this.state.description}
                        onChange={e => this.setState({description: e.target.value})}/>
                </form>
            </div>
        );
    }
}

export default AddTodo;