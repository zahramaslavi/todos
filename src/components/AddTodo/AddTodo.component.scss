.addTodo {
    padding-bottom: 1em;

    form {
        display: flex;
        flex-direction: column;
        text-align: left;

        > * {
            margin: .5em;
        }

        input {
            border-width: 0;
        }
    }

}