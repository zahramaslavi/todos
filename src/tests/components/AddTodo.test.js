import React from 'react';
import { shallow } from 'enzyme';
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";

import AddTodo from '../../components/AddTodo/AddTodo.component';

test('should test AddTodo component', () => {
    const wrapper = shallow(<AddTodo/>);
    expect(wrapper).toMatchSnapshot();
});

test('should test AddTodo component handler', () => {
    const onSubmitSpy = jest.fn();
    const wrapper = shallow(<AddTodo addTodo={onSubmitSpy}/>);
    const form = wrapper.find('div').find('form');
    const input = form.find('input');

    input.simulate('change', { target: { value: 'test' } });
    form.simulate('submit');
    expect(onSubmitSpy).toBeCalledWith({description: 'test',completed: false});
});


///////////////SNAPSHOT WITHOUT enzyme
// let container = null;
// beforeEach(() => {
//     // setup a DOM element as a render target
//     container = document.createElement("div");
//     document.body.appendChild(container);
// });

// afterEach(() => {
//     // cleanup on exiting
//     unmountComponentAtNode(container);
//     container.remove();
//     container = null;
// });

// it("renders AddTodo component with or without a addTodo event", () => {
//     const onSubmitSpy = jest.fn();

//     act(() => {
//         render(<AddTodo addTodo={onSubmitSpy}/>, container);
//     });
//     expect(container.querySelector('div.addTodo')).toBeTruthy();
//     expect(container.querySelector('div.addTodo form')).toBeTruthy();
//     expect(container.querySelector('div.addTodo form input')).toBeTruthy();
//     expect(container.querySelector('div.addTodo form input').placeholder).toBe('Type to add new tasks');
//     expect(container.querySelector('div.addTodo form input').type).toBe('text');

//     const form = container.querySelector('div.addTodo form');
//     const input = container.querySelector('div.addTodo form input');
//     expect(input.value).toBe('');
// });