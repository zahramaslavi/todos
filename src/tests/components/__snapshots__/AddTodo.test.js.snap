// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test AddTodo component 1`] = `
<div
  className="addTodo"
>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Type to add new tasks"
      type="text"
      value=""
    />
  </form>
</div>
`;
